### DISCUSSION TITLE
Do I need to introspect changes from db in each environment like qa/prod for mssql-server?

### DISCUSSION THREAD

**Discussion by shankuljain** [prisma/prisma#10223] [Q&A]
While the overall model structure in schema.prisma remains same, I notices name of constraints which user didn't define (like primary key) change with different DB servers. Can those hashes be safely ignored and Can I use same schema.prisma file across multiple db server environments?

**Comment by janpio** [comment]
With Prisma 3.x those constraint names should be included in your schema. Now if you create a new database (e.g. qa or staging environment) with Prisma, it should use the same constraint names if you create that database schema with `prisma migrate` or `prisma db push`.

Keeping these values different across environments can cause trouble - because in theory there could be other applications that rely on these name in some form.

Alternatively you can make Prisma switch the database to its inbuilt defaults: https://www.prisma.io/docs/guides/upgrade-guides/upgrading-versions/upgrading-to-prisma-3/named-constraints#option-2-i-want-to-use-prismas-default-constraint-and-index-names

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
